.github/workflow/workflow.yml

```diff
@@ -0,0 +1,59 @@
+name: python-deployment
+run-name: ${{ github.actor }} - ${{ github.ref_name }} - ${{ github.sha }}
+
+on:
+  push:
+    branches: ["main", "develop"]
+  pull_request:
+    branches: ["main"]
+
+env:
+  PYTHON_VERSION: 3.8
+
+jobs:
+  python-ci:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./src
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+
+        name: Setup Python ${{ env.PYTHON_VERSION }}
+        with:
+          python-version: "${{ env.PYTHON_VERSION }}"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest pytest-cov
+          pip install -r requirements.txt
+
+      - name: Lint
+        run: |
+          cd ..
```

Comment:
why is needed to exit from src to higher level dir if there aren't any other files to check